controllers/etcd_custodian_controller.go

```diff
@@ -168,6 +176,92 @@ func (ec *EtcdCustodian) updateEtcdStatus(ctx context.Context, logger logr.Logge
 	})
 }
 
+func calculatePDBminAvailable(etcd *druidv1alpha1.Etcd) int {
```

Comment:
Since we have many cases here it'd be awesome to have proper unit tests for this function.

Reply:
since the implementation is in the correct spot now, i will write some tests for this :)